### PR TITLE
Import Keystores Standard API Implementation

### PR DESCRIPTION
**Pull request by rauljordan**
This PR implements the `ImportKeystores` standard API endpoint defined in [here](https://ethereum.github.io/keymanager-APIs/#/) as a POST request. It allows for importing slashing protection, keystores with different passwords, and aligns with the specification. We already had an API endpoint named `ImportKeystores` as part of the Prysm endpoints. This is only used by the web interface, and we are safe to rename it to something more-specific to Prysm.

- [x] Rename current Prysm endpoint to `ImportAccounts` to reserve `ImportKeystores` for the standard API
- [x] Define an `Importer` interface that both the derived and imported keymanager implement
- [x] Amend the keymanager logic to support multiple passwords per keystore
- [x] Implement the API endpoint for `ImportKeystores` according to specification, ensuring it works as expected with necessary tests

**Note**: Diff large due to protobuf regeneration.

### CODE REVIEW

File: validator/accounts/accounts_import_test.go [hunk 1]
```diff
@@ -131,33 +131,6 @@ func TestImport_DuplicateKeys(t *testing.T) {
 	assert.Equal(t, 1, len(keys))
 }
 
-// TestImport_NonImportedWallet is a regression test that ensures non-silent failure when importing to non-imported wallets
```
Comment (rauljordan) [inline validator/accounts/accounts_import_test.go:134]: Derived wallets should be able to import keystores as they are merely wrappers over imported wallets

File: validator/rpc/wallet_test.go [hunk 2]
```diff
@@ -311,30 +311,7 @@ func TestServer_WalletConfig(t *testing.T) {
 	})
 }
 
-func TestServer_ImportKeystores_FailedPreconditions_WrongKeymanagerKind(t *testing.T) {
```
Comment (rauljordan) [inline validator/rpc/wallet_test.go:314]: Derived wallets should be able to import keystores as they are merely wrappers over imported wallets

**Review by mrcali213 — DISMISSED** [review]

File: validator/rpc/standard_api.go [hunk 3]
```diff
@@ -39,6 +40,41 @@ func (s *Server) ListKeystores(
...
+			ctx, s.valDB, bytes.NewBuffer([]byte(req.SlashingProtection)),
+		); err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not import slashing protection JSON: %v", err)
+		}
+	}
+	statuses, err := importer.ImportKeystores(ctx, keystores, req.Passwords)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not import keystores: %v", err)
+	}
+	return &ethpbservice.ImportKeystoresResponse{Statuses: statuses}, nil
```
Comment (james-prysm) [inline validator/rpc/standard_api.go:75]: we can also include the message with the status, but I need to ask about including the public key here too in the response...

File: validator/keymanager/imported/import.go [hunk 4]
```diff
@@ -20,26 +20,61 @@ import (
...
+		privKeyBytes, pubKeyBytes, _, err = km.attemptDecryptKeystore(decryptor, keystores[i], passwordToUse)
 		if err != nil {
-			return err
+			statuses[i] = &ethpbservice.ImportedKeystoreStatus{
+				Status:  ethpbservice.ImportedKeystoreStatus_ERROR,
+				Message: err.Error(),
+			}
+			continue
 		}
 		// if key exists prior to being added then output log that duplicate key was found
```
Comment (rkapka) [inline validator/keymanager/imported/import.go:60]: This comment should be moved lower.

File: validator/keymanager/imported/import.go [hunk 5]
```diff
@@ -20,26 +20,61 @@ import (
...
+		if err := bar.Add(1); err != nil {
+			statuses[i] = &ethpbservice.ImportedKeystoreStatus{
+				Status:  ethpbservice.ImportedKeystoreStatus_ERROR,
+				Message: err.Error(),
+			}
+			continue
```
Comment (rkapka) [inline validator/keymanager/imported/import.go:66]: If the progress bar is broken, let's just continue. I don't see a relationship between the progress bar failing and the import failing.

File: validator/rpc/standard_api.go [hunk 6]
```diff
@@ -39,6 +40,41 @@ func (s *Server) ListKeystores(
...
+	importer, ok := s.keymanager.(keymanager.Importer)
+	if !ok {
+		return nil, status.Error(codes.Internal, "Keymanager kind cannot import keys")
+	}
+	keystores := make([]*keymanager.Keystore, len(req.Keystores))
+	for i := 0; i < len(req.Keystores); i++ {
+		k := &keymanager.Keystore{}
+		if err := json.Unmarshal([]byte(req.Keystores[i]), k); err != nil {
+			return nil, status.Errorf(
+				codes.Internal, "Invalid keystore at index %d in request: %v", i, err,
```
[the remark this answers is not in the document]
Comment (rkapka) [reply validator/rpc/standard_api.go:59]: +1 for bad request.

**Review by rkapka — CHANGES_REQUESTED** [review]

File: validator/rpc/standard_api.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (rkapka) [reply validator/rpc/standard_api.go:59]: +1 for bad request

**Comment by rkapka** [comment]
One side-question: What happens when the user abruptly stops the import in the middle, e.g. by pressing Ctrl+C? Will they keystore become corrupted?

File: validator/rpc/standard_api.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (rauljordan) [reply validator/rpc/standard_api.go:59]: The API spec only allows for returning a few error codes, 200, 401, 403, and 500 and nothing else, unfortunately

**Comment by rauljordan** [comment]
> One side-question: What happens when the user abruptly stops the import in the middle, e.g. by pressing Ctrl+C? Will the keystore become corrupted?

Probably, but that risk exists anywhere we are writing to the DB, so there is no real good way to prevent this if a user forcefully kills a process

File: validator/keymanager/imported/import.go (hunk 5 above)
Comment (rauljordan) [reply validator/keymanager/imported/import.go]: Agreed!

File: validator/rpc/standard_api.go (hunk 3 above)
Comment (rauljordan) [reply validator/rpc/standard_api.go:75]: Let's revisit in a future PR

**Review by rkapka — APPROVED** [review]

**Pull request merged**